File: csharp/phone-number/PhoneNumber.cs
using System;
using System.Text.RegularExpressions;

public class PhoneNumber
{
    public static string Clean(string phoneNumber)
    {
        string number = Regex.Replace(phoneNumber, "[^0-9]", "");

        if (number.Length > 11 
            || number.Length < 10
            || number[number.Length - 10] < '2'
            || number[number.Length - 7] < '2'
            || (number.Length == 11 && number[0] != '1')
           )
        {
            throw new ArgumentException("Invalid phone number length");
        }

        return number.Length == 11 ? number.Substring(1) : number;
    }
}

File: python/phone-number/phone_number.py
import re


class PhoneNumber:
    def __init__(self, number):
        self.number = self.clean_number(number)
        self.area_code = self.number[0:3]

    def clean_number(self, number):
        number = re.sub("[^0-9]", "", number)
        if (
            len(number) > 11
            or len(number) < 10
            or number[len(number) - 10] < "2"
            or number[len(number) - 7] < "2"
            or len(number) == 11 and number[0] != "1"
        ):
            raise ValueError("Invalid number")

        return number[1:] if len(number) == 11 else number

    def pretty(self):
        return f"({self.area_code}) {self.number[3:6]}-{self.number[6:]}"
